File: ios/Classes/AudioFormat.swift
import Foundation

enum AudioFormat: String, CaseIterable {
    
    case AAC
    case WAV
    
    var ext: String {
        get {
            switch self {
            case .WAV:
                return ".wav"
            default://AAC
                return ".m4a"
            }
        }
    }
}

extension String? {
    func toAudioFormat() -> AudioFormat? {
        var cases = AudioExtension.allCases;
        var first = cases.first { $0.ext == self }
        var audioFormat = first?.audioFormat
        
        return AudioExtension.allCases.first { $0.ext == self }?.audioFormat
    }
}

File: android/src/main/kotlin/com/tcubedstudios/flutter_audio_recorder4/AudioFormat.kt
package com.tcubedstudios.flutter_audio_recorder4

import com.tcubedstudios.flutter_audio_recorder4.AudioExtension.Companion.toAudioExtension

enum class AudioFormat(val extension: String) {
    AAC(".m4a"),
    WAV(".wav");

    companion object {
        fun String?.toAudioFormat() : AudioFormat? {
            return toAudioExtension()?.audioFormat
        }
    }
}